taxi/tests/test_views.py

```diff
@@ -0,0 +1,111 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Driver, Car
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+DRIVER_URL = reverse("taxi:driver-list")
+CAR_URL = reverse("taxi:car-list")
+
+
+class PublicTest(TestCase):
+    def test_login_required_for_manufacturer_list_page(self):
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_for_driver_list_page(self):
+        response = self.client.get(DRIVER_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_for_car_list_page(self):
+        response = self.client.get(CAR_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PrivateTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        Manufacturer.objects.create(name="BMW", country="Germany")
+        Manufacturer.objects.create(name="Volvo", country="Sweden")
+        cls.car1 = Car.objects.create(
+            model="M4",
+            manufacturer=Manufacturer.objects.get(id=1)
+        )
+        cls.car2 = Car.objects.create(
+            model="XC60",
+            manufacturer=Manufacturer.objects.get(id=2)
+        )
+        Driver.objects.create(
+            username="test_user_1",
+            password="Test1234!",
+            license_number="AAA00001"
+        )
+        Driver.objects.create(
+            username="test_user_2",
+            password="Test1234!",
+            license_number="AAA00002"
+        )
```

Comment:
In the `setUpTestData` method, the creation of the Manufacturer, Car, and Driver objects is hardcoded with specific IDs. This may lead to problems if the IDs are not available or have been used elsewhere in the database. It's recommended to avoid hardcoding IDs and instead assign the created objects to variables, which can be used later.